src/pivot.coffee

```diff
@@ -660,11 +661,10 @@ callWithJQuery ($) ->
 
                     #Only allow clicking on the finest-grained attribute.
                     if getHeaderClickHandler? and colAttrIdx == (colAttrs.length - 1)
-                        flatColKey = colKey.join(String.fromCharCode(0))
+                        flatColKey = colKey.join(FLAT_KEY_DELIM)
                         th.onclick = getHeaderClickHandler("col", "key", flatColKey)
-                        #Add key to data-set for post-processing sort icons. CSS selectors
-                        #don't work w/code point 0, but we don't need the separators anyway.
-                        th.setAttribute("data-flat-key", colKey.join(""))
+                        #Add key to data-set for post-processing sort icons.
+                        th.setAttribute("data-flat-key", flatColKey)
```

Comment:
See corresponding change in PivotTable.js:syncSortIcons().

We set the flatKey data attribute on the header cell that was clicked so that we can find it and add a sort icon to it after refreshing the table with the new sort. Apparently the null byte can't be easily used in CSS selectors, so what I did was ignore the delimiter when setting this data attribute, since it's not necessary for this workflow.